src/services/pegout-data.processor.ts

```diff
@@ -1,27 +1,70 @@
 import { BridgeDataFilterModel } from '../models/bridge-data-filter.model';
 import { PegoutStatusDataService } from './pegout-status-data-services/pegout-status-data.service';
+import { PegoutStatusDataModel } from '../models/rsk/pegout-status-data.model';
+import { PegoutStatusRulesService } from './pegout-status/pegout-status-rules-services';
+import {ServicesBindings} from '../dependency-injection-bindings';
 
 export class PegoutDataProcessor implements FilteredBridgeTransactionProcessor {
-  logger: Logger;
+  PegoutStatusRulesService: PegoutStatusRulesService;
   pegoutStatusDataService: PegoutStatusDataService;
+  logger: Logger;
 
-  constructor(pegoutStatusDataService: PegoutStatusDataService) {
+  constructor(
+    @inject(ServicesBindings.PEGOUT_STATUS_SERVICE)
+    PegoutStatusRulesService: PegoutStatusRulesService,
+    pegoutStatusDataService: PegoutStatusDataService)
+ {
     this.logger = getLogger('pegoutDataProcessor');
     this.pegoutStatusDataService = pegoutStatusDataService;
+    this.PegoutStatusRulesService = PegoutStatusRulesService;
   }
   async process(rskTransaction: RskTransaction): Promise<void> {
-    this.logger.error('[process] Method not yet implemented');
-    throw new Error('Method not yet implemented');
+    this.logger.debug(`[process] Got tx ${rskTransaction.hash}`);
+    const pegoutStatus = this.parse(rskTransaction);
+    if (!pegoutStatus) {
+      this.logger.debug('[process] Transaction is not a peg-out');
+      return;
+    }
+
+    try {
+      const found = await this.pegoutStatusDataService.getById(pegoutStatus.rskTxHash);
+      if (found) {
+        return this.logger.debug(`[process] ${rskTransaction.hash} already registered`);
+      }
+      await this.pegoutStatusDataService.set(pegoutStatus);
+      this.logger.trace(`[process] ${rskTransaction.hash} registered`);
+    } catch (e) {
+      this.logger.warn('[process] There was a problem with the storage', e);
+    }
   }
+
   getFilters(): BridgeDataFilterModel[] {
-    // TODO: add BRIDGE_METHODS.RELEASE_BTC = 'releaseBtc' when this method is available in the bridge abis.
     return [
       new BridgeDataFilterModel(getBridgeSignature(BRIDGE_METHODS.UPDATE_COLLECTIONS)),
       BridgeDataFilterModel.EMPTY_DATA_FILTER
     ];
   }
+
+  parse(transaction: RskTransaction): PegoutStatusDataModel | null {
+    if (!transaction || !transaction.logs || !transaction.logs.length) {
+      this.logger.warn(`[parse] This transaction doesn't have the data required to be parsed`);
+      return null;
+    }
+    const result = this.PegoutStatusRulesService.searchStatus(transaction);
+    if (!result) {
+      return null;
+    }
+    result.originatingRskTxHash = transaction.hash;
+    result.rskBlockHeight = transaction.blockHeight;
+    result.lastUpdatedOn = transaction.createdOn;
+    result.rskTxHash = transaction.hash; // this.getbtcTxId(transaction.data);
```

Comment:
as a general rule, try to avoid leaving commented code, but reading the commented code I don't understand why you would be trying to get the btc tx id here.

Reply:
comment removed.